test/resources/examples/index.js

```diff
@@ -0,0 +1,144 @@
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(ERROR_EPSILON), user: 3 },
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(20)).add(ERROR_EPSILON), buyAmount: toETH(10), user: 4 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(ERROR_EPSILON), user: 5 },
+  ],
+  solutions: [
+    {
+      name: "Match 1 Pair",
+      prices: [1, 2].map(toETH),
+      buyVolumes: [toETH(10), feeSubtracted(toETH(20)), ZERO, ZERO, ZERO, ZERO]
+    },
+    {
+      name: "Match 2 Pairs",
+      prices: [1, 2].map(toETH),
+      buyVolumes: [toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20)), ZERO, ZERO]
+    },
+    {
+      name: "Match 3 Pairs",
+      prices: [1, 2].map(toETH),
+      buyVolumes: [toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20))]
+    },
+  ],
+})
+
+const biggieSmallTrade = generateTestCase({
+  name: "Biggie Small",
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: toETH(185), buyAmount: toETH(1), user: 0 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(1000), buyAmount: toETH(184000), user: 1 },
+  ],
+  solutions: [
+    {
+      name: "Max Fulfillment",
+      prices: [toETH(1), feeAdded(toETH(184))],
+      buyVolumes: [1, 184].map(toETH),
+    },
+  ],
+})
+
+const basicRingTrade = generateTestCase({
+  name: "Basic Ring",
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 0 },
+    { sellToken: 1, buyToken: 2, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 1 },
+    { sellToken: 2, buyToken: 0, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 2 },
+  ],
+  solutions: [
+    {
+      name: "Ring Around the Rosie",
+      prices: [1, 1, 1].map(toETH),
+      buyVolumes: [feeSubtracted(toETH(1), 1), feeSubtracted(toETH(1), 2), feeSubtracted(toETH(1), 3)],
+    },
+  ],
+})
+
+const shortRingBetterTrade = generateTestCase({
+  orders: [
+    // ring trade orders
+    { sellToken: 0, buyToken: 1, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 0 },
+    { sellToken: 1, buyToken: 2, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 1 },
+    { sellToken: 2, buyToken: 0, sellAmount: toETH(1), buyAmount: toETH(0.99), user: 2 },
+    // biggie small orders
+    { sellToken: 0, buyToken: 1, sellAmount: toETH(185), buyAmount: toETH(1), user: 3 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(1000), buyAmount: toETH(184000), user: 4 },
+  ],
+  solutions: [
+    {
+      name: "Ring Trade Solution",
+      prices: [1, 1, 1].map(toETH),
+      buyVolumes: [feeSubtracted(toETH(1), 1), feeSubtracted(toETH(1), 2), feeSubtracted(toETH(1), 3), ZERO, ZERO],
+    },
+    {
+      name: "Biggie Small Trade",
+      prices: [toETH(1), feeAdded(toETH(184)), ZERO],
+      buyVolumes: [ZERO, ZERO, ZERO, toETH(1), toETH(184)],
+    },
+  ],
+})
+
+const fiveThouzy = new BN("5000")
```

Comment:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2F3NeRncMrUNb8astzVy%2Fgiphy.gif&token=joker" data-gitmeme-token="joker"><img src="https://media1.giphy.com/media/3NeRncMrUNb8astzVy/giphy.gif" title="Created by gitme.me with /joker"/></a>

Reply:
Glad you like it.

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FO5xChSjqUIxsk%2Fgiphy.gif&token=not" data-gitmeme-token="not"><img src="https://media3.giphy.com/media/O5xChSjqUIxsk/giphy.gif" title="Created by gitme.me with /not"/></a>

Reply:
I suppose you'd like me to change it. The most appropriate place would be here:

#301